### ISSUE TITLE
BIND 9.16 compatibility

### ISSUE THREAD

**Issue by sthen** [DNS-OARC/dnsperf#67]
BIND 9.16 removes isc-config.sh and changes API of isc_mem_create() and isc_buffer_allocate(). I don't have time to do a full PR-ready diff with autoconf checks to make it work with either version, but to hopefully save someone else a bit of time, this is what I'm using in the OpenBSD port for now.

The API changes were in https://github.com/isc-projects/bind9/commit/1b716a39f51200027388d90290488a648a8ccf5b and https://github.com/isc-projects/bind9/commit/4459745ff2425f38418acd5be2fcaf9bc969fc61

``` 
Index: src/dns.c
--- src/dns.c.orig
+++ src/dns.c
@@ -137,10 +137,7 @@ perf_dns_createctx(bool updates)
         return NULL;
 
     mctx   = NULL;
-    result = isc_mem_create(0, 0, &mctx);
-    if (result != ISC_R_SUCCESS)
-        perf_log_fatal("creating memory context: %s",
-            isc_result_totext(result));
+    isc_mem_create(&mctx);
 
     ctx = isc_mem_get(mctx, sizeof(*ctx));
     if (ctx == NULL) {
@@ -373,9 +370,7 @@ perf_dns_parseednsoption(const char* arg, isc_mem_t* m
 
     option->mctx   = mctx;
     option->buffer = NULL;
-    result         = isc_buffer_allocate(mctx, &option->buffer, strlen(value) / 2 + 4);
-    if (result != ISC_R_SUCCESS)
-        perf_log_fatal("out of memory");
+    isc_buffer_allocate(mctx, &option->buffer, strlen(value) / 2 + 4);
 
     result = isc_parse_uint16(&code, copy, 0);
     if (result != ISC_R_SUCCESS) {

Index: src/dnsperf.c
--- src/dnsperf.c.orig
+++ src/dnsperf.c
@@ -389,10 +389,7 @@ setup(int argc, char** argv, config_t* config)
     isc_result_t result;
     const char*  mode = 0;
 
-    result = isc_mem_create(0, 0, &mctx);
-    if (result != ISC_R_SUCCESS)
-        perf_log_fatal("creating memory context: %s",
-            isc_result_totext(result));
+    isc_mem_create(&mctx);
 
     dns_result_register();
 
Index: src/resperf.c
--- src/resperf.c.orig
+++ src/resperf.c
@@ -226,10 +226,7 @@ setup(int argc, char** argv)
     isc_result_t result;
     const char*  _mode = 0;
 
-    result = isc_mem_create(0, 0, &mctx);
-    if (result != ISC_R_SUCCESS)
-        perf_log_fatal("creating memory context: %s",
-            isc_result_totext(result));
+    isc_mem_create(&mctx);
 
     dns_result_register();
 
```

**Comment by jelu** [comment]
Thanks for the patch but with the recent changes to BIND, specifically the statement that their libraries should not be used outside of BIND and the removal of `isc-config.sh`, I think we need to remove this dependency completely...

**Comment by neirbowj** [comment]
Heads-up: dnsperf was [removed from the FreeBSD ports tree](https://svnweb.freebsd.org/changeset/ports/533423) when BIND 9.14 expired on 04/30.

**Comment by jelu** [comment]
@neirbowj Thanks for the info, sadly it's not an easy fix...

**Comment by neirbowj** [comment]
Should I be able to build against BIND 9.11? That might be an easy way to get dnsperf back into the ports tree for the time being.

**Comment by sthen** [comment]
Yes, or my patches do work (you'll need to zap the autoconf check for isc-config and just use -lbind9 instead of isc-config --libs).

**Comment by jelu** [comment]
@neirbowj Yes, you can build against 9.11

@sthen Well, that is only a quick fix in the short run. ISC have stated that their libraries are not for use outside of BIND.

**Comment by pemensik** [comment]
Hi. I have modified those changes to autodetection. This way, it is still able to compile with BIND 9.11, but also with BIND 9.16. It does still depend on bind libraries however.

**Comment by pemensik** [comment]
It seems to me I solved also issue #23, because I hit error in Fedora with symbol missing symbol ERR_get_error. -lcrypto seems mandatory with OpenSSL 1.1

**Comment by jelu** [comment]
THANK YOU @pemensik !!!! :D
